SD_Assignment1/src/main/java/ro/utcn/aron/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,409 @@
+	}
+
+	@Override
+	public List<Question> filterByTag(String tag) {
+		return template.query(
+				"SELECT * FROM question JOIN tags ON question.id = tags.questionid WHERE tags.tags LIKE '%" + tag
+						+ "%'",
+				(resultSet, i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"),
+						resultSet.getString("body"), list2String(getTagsFromJdbc(resultSet.getInt("id"))),
+						resultSet.getString("author"), resultSet.getString("author"),getAnswersFromJdbc(resultSet.getInt("id")), getQuestionScore(resultSet.getInt("id"))));
+	}
+
+	private List<String> getTagsFromJdbc(int questionId) {
+		return template.query("SELECT * FROM tags WHERE questionid = ?",
+				(resultSet, i) -> new String(resultSet.getString("tags")), questionId);
+	}
+
+	private String list2String(List<String> tagList) {
+		String text = "";
+		for (int i = 0; i < tagList.size() - 1; i++) {
+			text += tagList.get(i) + ",";
+		}
+		text += tagList.get(tagList.size() - 1);
+
+		return text;
+	}
+
+	@Override
+	public void answerQuestion(int questionid, String user, String answer) {
+		SimpleJdbcInsert insert = new SimpleJdbcInsert(template);
+		insert.setTableName("answers");
+		insert.setGeneratedKeyName("answerid");
+
+		Map<String, Object> data = new HashMap<>();
+		data.put("questionid", questionid);
+		data.put("author", user);
+		data.put("answer", answer);
+
+		data.put("creationdate", new Date().toString());
+		insert.execute(data);
+	}
+
+	private List<Answer> getAnswersFromJdbc(int questionid) {
+		return template.query("SELECT * FROM answers WHERE questionid = ?",
+				(resultSet, i) -> new Answer(resultSet.getInt("answerid"),resultSet.getString("answer"), resultSet.getString("author"),
+						resultSet.getString("creationdate"), getAnswerScore(resultSet.getInt("answerid")) ),
+				questionid);
+	}
+	
+	private int getAnswerScore(int answerid) {
+		return template.query("SELECT * FROM avotes WHERE answerid = ?",
+				(resultSet, i) -> new Vote(resultSet.getString("author"), resultSet.getInt("voteid"), resultSet.getInt("votetype")),
+				answerid).stream().mapToInt(vote->vote.getVoteYype()).sum();
+	}
+	
+	private int getQuestionScore(int questionid) {
+		return template.query("SELECT * FROM qvotes WHERE questionid = ?",
+				(resultSet, i) -> new Vote(resultSet.getString("author"), resultSet.getInt("voteid"), resultSet.getInt("votetype")),
+				questionid).stream().mapToInt(vote->vote.getVoteYype()).sum();
+	}
+
+	@Override
+	public void editQuestion(int questionid, String user, String text) {
+		List<Question> result = template.query("SELECT * FROM question WHERE id = ? and author = ?", (resultSet,
+				i) -> new Question(resultSet.getInt("id"), resultSet.getString("title"), resultSet.getString("body"),
+						list2String(getTagsFromJdbc(resultSet.getInt("id"))), resultSet.getString("author"),
+						resultSet.getString("creationdate"),getAnswersFromJdbc(resultSet.getInt("id"))), questionid, user);
+		
+		if(result.isEmpty()) {
```

Comment:
Again, this is business logic:
 - You should have moved this to the service layer (as you partially replicate this code in the `InMemoryQuestionRepository`).
 - You should not use `System.out.println` in this layer (or the service layer) at all. We do not want to depend on the user interface at this point, but we want to be able to swap it easily. I would suggest that you define Exceptions instead of these println calls, catch them in the console controller and then print corresponding error messages on the command line.